pkgs/data/themes/colloid/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, gtk3
+, gnome-themes-extra
+, gtk-engine-murrine
+, sassc
+, tweaks ? [ ] 
+, jdupes
+, bash
+}:
+let
+  validTweaks = [ "compact" "nord" "rimless" "normal" "dracula" ];
+  unknownTweaks = lib.subtractLists validTweaks tweaks;
+in
+  assert lib.assertMsg (unknownTweaks == [ ]) ''
+    You entered wrong tweaks: ${toString unknownTweaks}
+    Valid tweaks are: ${toString validTweaks}
+  '';
+
+  stdenvNoCC.mkDerivation rec {
+    pname = "Colloid-gtk-theme";
+    version = "2022-05-15";
+
+    src = fetchFromGitHub {
+      owner = "vinceliuice";
+      repo = pname;
+      rev = "2c674203ab19776dd8b8dd74d88e239d89fbcf61";
+      sha256 = "pdc5lRFT3xiLCGLxhBXTUGkbqEqr5/e3nrQrlx2qIOg=";
+      name = "pname";
+      };
+
+  nativeBuildInputs = [ jdupes sassc ];
+  buildInputs = [ bash gnome-themes-extra ];
+  propagatedUserEnvPkgs = [ gtk-engine-murrine ];
+
+  preInstall = ''
+    mkdir -p $out/share/themes
+    patchShebangs ./install.sh ./gtkrc.sh ./build.sh ./clean-old-theme.sh
+  '';
+
+  installPhase = ''
+    runHook preInstall
+    HOME=./tmp
+    bash install.sh -d $out/share/themes -t all ${lib.optionalString (tweaks != []) "--tweaks " + builtins.toString tweaks}
+    runHook postInstall
+  '';
+
+  postInstall = ''
+    jdupes -L -r $out/share
```

Comment:
Why is this necessary? If their build script is creating duplicates, that should be fixed upstream.

Reply:
Hey! Sorry for keeping this PR open for no reason. I had to deal with real life things and forgot it entirely. There is already an alternative package for this theme in nixpkgs, so there is no need to keep this one around. I believe the reason I added the postInstall is only in the case of installing multiple copies of the theme, in which case some redundant files are removed. I may be wrong, but I believe it should not be applied at the original install script, as it is not opinionated about install locations and other configuration details, whereas with NixOS details about the installation can be reliably made. It should be possible to upstream changes to the install script, but those concerns should be directed to the maintainer of the current nixpkgs package, as it appears to use jdupes as well.
Regarding the depency on colloid icon theme. I believe it is only for the best visual experience and the theme works with alternative icon sets without issues at least in my personal use.